auth0/resource_auth0_global_client.go

```diff
@@ -0,0 +1,43 @@
+package auth0
+
+import (
+	"errors"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"gopkg.in/auth0.v3"
+	"gopkg.in/auth0.v3/management"
+	"net/url"
+)
+
+func newGlobalClient() *schema.Resource {
+	client := newClient()
+	client.Create = createGlobalClient
+	client.Delete = deleteGlobalClient
+
+	name := client.Schema["name"]
+	name.Required = false
+	name.Computed = true
+
+	return client
+}
+
+func createGlobalClient(d *schema.ResourceData, m interface{}) error {
+	api := m.(*management.Management)
+	clients, err := api.Client.List(withIsGlobal(), management.WithFields("client_id"))
+	if err != nil {
+		return err
+	} else if len(clients.Clients) == 0 {
+		return errors.New("no auth0 global client found")
+	}
+	d.SetId(auth0.StringValue(clients.Clients[0].ClientID))
```

Comment:
You could also do `d.SetId(clients.Clients[0].GetClientID())`. I've added generation of getters in go-auth0/auth0 recently.

Reply:
done